Global.h

```diff
@@ -576,24 +577,24 @@ namespace vstd
 		return i >= 0  &&  i < c.size();
 	}
 
-	template <typename Container, typename Index>
-	boost::optional<typename Container::const_reference> tryAt(const Container &c, Index i)
+	template<typename Container, typename Index>
+	std::optional<typename Container::value_type> tryAt(const Container & c, Index i)

```

Comment:
Why const_ref has been changed to value_type and only here?
Similar methods don't have similar change

Reply:
@IvanSavenko because other similar methods are unused. Const_ref is not allowed in std::optional, and I tried but unsuccessful in using std::reference_wrapper here. Perhaps remove unused methods?

Reply:
In that case:
- remove unused methods
- optionally - remove this method (only 1 usage in code)

Reply:
unused functions removed, method converted to lambda and moved inside VCAI